### PR TITLE
Issue #5508: export/manifest/extraction mutation coverage (cheap-lane worker)

### PR DESCRIPTION
**Pull request by ll7**
## What / Why

This is the **second slice** of issue #5508, adding NEW capability on top of the merged first slice
(PR #5513, commit `9418c2dd`). It does not duplicate #5513: that slice covered `aggregate_metrics`,
`bootstrap_ci`, and `compute_completeness_score`. This slice adds targeted tests for the 218 `no-test`
mutants that the first slice explicitly left untracked, covering:

- `export_metrics_json`
- `export_metrics_csv`
- `_load_manifest_payload`
- `extract_seed_metrics`

## Changes

- `tests/research/test_aggregation.py`: 12 new tests (total now 28) exercising JSON/CSV round-trips,
  parent-directory creation, JSONL last-non-empty-line semantics, empty/malformed manifest handling,
  missing/non-numeric metrics, and missing-file failures in `extract_seed_metrics`.
- `mutation_testing_triage.md`: updated untested-functions section to record the new coverage and
  re-state the still-open scheduled-diagnostic-job task.

## Validation Output

- `uv run pytest tests/research/test_aggregation.py` — 28 passed.
- `uv run ruff check tests/research/test_aggregation.py` — passed.
- `uv run ruff format --check tests/research/test_aggregation.py` — passed.

No campaigns, Slurm, training, or benchmark/paper claims. CPU-level unit validation only.

## Research Result Guidance

- Target claim / hypothesis / blocker: add targeted unit-test coverage for four aggregation export,
  manifest, and seed-extraction paths; no mutation-score claim is made.
- Comparator or baseline: the first #5513 slice and the four previously untested functions listed in
  `mutation_testing_triage.md`.
- Evidence tier: diagnostic unit-test coverage.
- Result classification: test-hardening / partial mutation-coverage progress.
- Decision or stop rule: stop after the focused 28-test suite passes; scheduled mutmut automation
  remains on #5508 and is not a required PR gate.
- New research/benchmark/metric/paper-facing analysis tool: no.

## Domain-Aware Approval

- Required for this PR: yes — the tests target benchmark-critical aggregation/provenance semantics,
  but no scientific result is interpreted.
- Domains reviewed: metric aggregation, manifest/provenance parsing, failure classification, claim
  boundary.
- Status: approved for implementation integrity; scientific interpretation waived.
- Approver/review source or waiver: issue #5508 contract plus this gate’s review of record.
- Validity checklist:
  - Target claim/hypothesis: covered failure and round-trip paths behave as asserted; no outcome claim.
  - Comparator or split/evidence validity: assertions exercise the existing function contracts and
    distinguish valid records from structured failures.
  - Fallback/degraded exclusions: no benchmark rows or fallback/degraded runs are promoted.
  - Claim boundary: diagnostic CPU-level test coverage only; no paper, leaderboard, or mutation-score
    claim.
  - Implementation integrity vs experimental validity: tests prove code-path behavior, not
    experimental validity.

## Successor / What Remains

Refs #5508

Remaining task (tracked by #5508):
- A scheduled (weekly/nightly) `mutmut` diagnostic job producing a surviving-mutant report and
  alerting on regressions. This PR does not implement that workflow.

## Follow-Up Issues

- #5508 remains open for the scheduled mutmut diagnostic workflow and the remaining no-test mutant
  category; this PR is a partial slice and does not close the parent issue.

This PR was produced by the cheap implementation lane; the review gate hardens and merges.

### CODE REVIEW

**Comment by ll7** [comment]
## Gate findings — phase 1 review

Reviewed the actual diff at head `4b995ae6c1a7e046192fae0cf9b2619cf164d9df`, linked issue #5508 with comments, and the existing current-main checks.

**Open review threads: 4**

1. **`discussion_r3572418474` — `tests/research/test_aggregation.py:405` (Gemini MEDIUM):** the missing-metrics failure test does not fully assert `policy_type` or use diagnostic truthiness/assertion messages. **Verdict: fixed in phase 2** with complete failure-payload assertions and messages.
2. **`discussion_r3572418487` — `tests/research/test_aggregation.py:418` (Gemini MEDIUM):** the non-numeric-metrics failure test omits `path`, `seed`, and `policy_type` validation. **Verdict: fixed in phase 2** with complete payload assertions and messages.
3. **`discussion_r3572418495` — `tests/research/test_aggregation.py:430` (Gemini MEDIUM):** the malformed-JSON failure test omits `path` and `policy_type` validation. **Verdict: fixed in phase 2** with complete payload assertions and messages.
4. **`discussion_r3572418503` — `tests/research/test_aggregation.py:439` (Gemini MEDIUM):** the missing-file failure test only checks that a reason exists. **Verdict: fixed in phase 2** with complete payload assertions, a specific missing-file reason check, and messages.

**Additional gate findings: 0.** The existing test file uses `tmp_path`, the PR correctly uses `Refs #5508`, and the remaining scheduled mutmut workflow is explicitly tracked on open issue #5508. CodeRabbit’s two low-value optional nitpicks are not open review threads and are non-blocking; this gate is the review of record.

File: tests/research/test_aggregation.py [hunk 1]
```diff
@@ -268,3 +275,165 @@ def test_completeness_score_fail():
...
+    """Manifests without a metrics block are reported as failures."""
+    bad = tmp_path / "bad.json"
+    bad.write_text(json.dumps({"seed": 3, "policy_type": "baseline"}), encoding="utf-8")
+
+    records, failures = extract_seed_metrics([str(bad)])
+    assert records == []
+    assert len(failures) == 1
+    assert failures[0]["path"] == str(bad)
+    assert failures[0]["seed"] == 3
+    assert "metrics not found" in failures[0]["reason"]
```
[the remark this answers is not in the document]
Comment (ll7) [reply tests/research/test_aggregation.py]: Fixed in 14a42fd6b: the missing-metrics test now asserts empty records, count, path, seed, policy_type, and reason with diagnostic messages.

File: tests/research/test_aggregation.py [hunk 2]
```diff
@@ -268,3 +275,165 @@ def test_completeness_score_fail():
...
+    """Manifests with non-numeric metrics only are failures."""
+    bad = tmp_path / "meta.json"
+    bad.write_text(
+        json.dumps({"seed": 5, "policy_type": "baseline", "metrics": {"policy_type": "baseline"}}),
+        encoding="utf-8",
+    )
+
+    records, failures = extract_seed_metrics([str(bad)])
+    assert records == []
+    assert failures[0]["reason"] == "no numeric metrics found"
```
[the remark this answers is not in the document]
Comment (ll7) [reply tests/research/test_aggregation.py]: Fixed in 14a42fd6b: the non-numeric-metrics test now validates the complete failure metadata and reason with diagnostic messages.

File: tests/research/test_aggregation.py [hunk 3]
```diff
@@ -268,3 +275,165 @@ def test_completeness_score_fail():
...
+def test_extract_seed_metrics_unparseable_jsonl_fails(tmp_path: Path):
+    """Malformed JSON lines are captured as parse failures."""
+    bad = tmp_path / "broken.jsonl"
+    bad.write_text("{not valid json}", encoding="utf-8")
+
+    records, failures = extract_seed_metrics([str(bad)])
+    assert records == []
+    assert len(failures) == 1
+    assert failures[0]["seed"] is None
+    assert "Expecting" in failures[0]["reason"]
```
[the remark this answers is not in the document]
Comment (ll7) [reply tests/research/test_aggregation.py]: Fixed in 14a42fd6b: the malformed-JSON test now validates path, seed, policy_type, and parse reason with diagnostic messages.

File: tests/research/test_aggregation.py [hunk 4]
```diff
@@ -268,3 +275,165 @@ def test_completeness_score_fail():
...
+
+
+def test_extract_seed_metrics_missing_file_fails(tmp_path: Path):
+    """A missing manifest path is captured as a failure, not a crash."""
+    missing = tmp_path / "does_not_exist.json"
+    records, failures = extract_seed_metrics([str(missing)])
+    assert records == []
+    assert len(failures) == 1
+    assert failures[0]["reason"]
```
[the remark this answers is not in the document]
Comment (ll7) [reply tests/research/test_aggregation.py]: Fixed in 14a42fd6b: the missing-file test now validates path, seed, policy_type, and the missing-file reason with diagnostic messages.

**Comment by ll7** [comment]
## Gate phase 2 delta — repair delivered

Head is now `97c009dd5518932af394ece23b6c31ec523d6283` after the failure-metadata assertions, body contract, and current-main sync. Commit `14a42fd6b` fixes all four phase-1 MEDIUM findings; focused aggregation validation passed 28 tests with Ruff clean. **Open review threads: 0.**
